internal/attestation/crafter/crafter.go

```diff
@@ -482,8 +482,12 @@ func (c *Crafter) AddMaterialContractFree(ctx context.Context, attestationID, ki
 		return fmt.Errorf("%q kind not found. Available options are %q", kind, schemaapi.ListAvailableMaterialKind())
 	}
 
-	// 2 - Generate a random name for the material
-	m.Name = fmt.Sprintf("material-%d", time.Now().UnixNano())
+	// 2 - Set the name of the material if provided
+	m.Name = name
+	if m.Name == "" {
+		// 2.1 - Generate a random name for the material since it was not provided
+		m.Name = fmt.Sprintf("material-%d", time.Now().UnixNano())
+	}
 
 	// 3 - Craft resulting material
 	return c.addMaterial(ctx, &m, attestationID, value, casBackend, runtimeAnnotations)
```

Comment:
now we need to be careful about conflicts in names. What happens if you try a material twice with the same name? Override?